File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.cpp
class Solution {
public:
    bool closeStrings(string word1, string word2) {
        vector<int> cnt1(26), cnt2(26);
        unordered_set<char> s1, s2;
        for(auto c : word1){
            cnt1[c - 'a']++;
            s1.insert(c);
        }
        for(auto c : word2){
            cnt2[c - 'a']++;
            s2.insert(c);
        }
        sort(begin(cnt1), end(cnt1));
        sort(begin(cnt2), end(cnt2));
        return cnt1 == cnt2 && s1 == s2;
    }
};

File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.py
class Solution:
    def closeStrings(self, w1: str, w2: str) -> bool:
        c1, c2 = Counter(w1),  Counter(w2)
        # print(c1, c1.keys(), c1.values())
        # print(c2, c2.keys(), c2.values())
        return sorted(c2.values()) == sorted(c1.values()) and c1.keys() == c2.keys()